source/common/network/dns_impl.cc

```diff
@@ -215,11 +220,15 @@ ActiveDnsQuery* DnsResolverImpl::resolve(const std::string& dns_name,
   }
 }
 
-void DnsResolverImpl::PendingResolution::getHostByName(int family) {
-  ares_gethostbyname(
-      channel_, dns_name_.c_str(), family,
-      [](void* arg, int status, int timeouts, hostent* hostent) {
-        static_cast<PendingResolution*>(arg)->onAresHostCallback(status, timeouts, hostent);
+void DnsResolverImpl::PendingResolution::getAddrInfo(int family) {
+  struct ares_addrinfo_hints hints = {};
+  hints.ai_family = family;
+  hints.ai_flags = ARES_AI_CANONNAME | ARES_AI_ENVHOSTS | ARES_AI_NOSORT;
```

Comment:
Why `ARES_AI_CANONNAME`? Please add comments as appropriate for these flags.

Reply:
`ARES_AI_CANONNAME` is used for querying CNAME. Not sure if we need it.

Reply:
I don't think we need this, suggest dropping.